domains/client/block-preprocessor/src/lib.rs

```diff
@@ -329,31 +329,65 @@ where
             Block::Hash,
             Balance,
         >,
+        consensus_block_hash: CBlock::Hash,
         at: Block::Hash,
-    ) -> bool {
+    ) -> sp_blockchain::Result<bool> {
+        let bundle_vrf_hash =
+            U256::from_be_bytes(bundle.sealed_header.header.proof_of_election.vrf_hash());
+        let tx_range = self
+            .consensus_client
+            .runtime_api()
+            .domain_tx_range(consensus_block_hash, self.domain_id)
+            .map_err(|error| {
+                sp_blockchain::Error::Application(Box::from(format!(
+                    "Error getting tx range: {error}"
+                )))
+            })?;
```

Comment:
If we can move this to `compile_own_domain_bundles` then we only need to query the `tx_range` once for all bundles, non-blocker.